### DISCUSSION TITLE
My HttpRouter which uses queryParam was denied by the admission webhook.

### DISCUSSION THREAD

**Discussion by Yelijah** [Kong/kubernetes-ingress-controller#5701] [Q&A] [answered]
I want to create a HttpRoute like:
```
$ kubectl apply -f /home/elijah/k8s/study_yaml/gateway-api/httproute-queryparam.yaml
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: echo-param1
  annotations:
    konghq.com/strip-path: 'true'
spec:
  parentRefs:
  - name: kong
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /echo
      queryParams:
      - name: "tenantId"
        type: "RegularExpression"
        value: "a"
    backendRefs:
    - name: echo1
      kind: Service
      port: 1027
``` 
But it was denied by admission webhook with error like:
```
Error from server: error when creating "/home/elijah/k8s/study_yaml/gateway-api/httproute-queryparam.yaml": admission webhook "validations.kong.konghq.com" denied the request: HTTPRoute spec did not pass validation: rules[0].matches[0]: queryparam matching is supported with expression router only
Error from server: error when creating "/home/elijah/k8s/study_yaml/gateway-api/httproute-queryparam.yaml": admission webhook "validations.kong.konghq.com" denied the request: HTTPRoute spec did not pass validation: rules[0].matches[0]: queryparam matching is supported with expression router only
Error from server: error when creating "/home/elijah/k8s/study_yaml/gateway-api/httproute-queryparam.yaml": admission webhook "validations.kong.konghq.com" denied the request: HTTPRoute spec did not pass validation: rules[0].matches[0]: queryparam matching is supported with expression router only

```
How to fix it?

**Comment by pmalek** [comment]
You're looking for the router flavor change: https://docs.konghq.com/kubernetes-ingress-controller/latest/upgrade/kic/#expression-router

**Reply by Yelijah** [reply]
Im' sorry that I'm new here. How can I use k8s gateway api whit queryParam. Is it not supported by kong ?

**Reply by pmalek** [reply]
KIC (https://github.com/Kong/kubernetes-ingress-controller/) generates Kong configuration based on its configuration https://docs.konghq.com/kubernetes-ingress-controller/latest/reference/cli-arguments/.

One of those is Kong's router flavor which you can set/override in our helm chart.

Can you describe how you've deployed Kong + KIC to your cluster?

**Reply by Yelijah** [reply]
I use the latest helm chart to deploy kic with custom values.yaml like:
```
gateway:
  proxy:
    type: NodePort
    http:
      nodePort: 31081
    tls:
      enable: false
```

Which flag in the document https://docs.konghq.com/kubernetes-ingress-controller/latest/reference/cli-arguments/ can I use to support queryParam?

**Comment by Yelijah** [comment, marked as answer]
> KIC (https://github.com/Kong/kubernetes-ingress-controller/) generates Kong configuration based on its configuration https://docs.konghq.com/kubernetes-ingress-controller/latest/reference/cli-arguments/.
> 
> One of those is Kong's router flavor which you can set/override in our helm chart.
> 
> Can you describe how you've deployed Kong + KIC to your cluster?



> KIC (https://github.com/Kong/kubernetes-ingress-controller/) generates Kong configuration based on its configuration https://docs.konghq.com/kubernetes-ingress-controller/latest/reference/cli-arguments/.
> 
> One of those is Kong's router flavor which you can set/override in our helm chart.
> 
> Can you describe how you've deployed Kong + KIC to your cluster?

I have found the right configuration in kic values.yaml:
```
gateway:
  env:
    router_flavor: "expressions"
```

Thank you for your help!
